readthedocs/vcs_support/backends/git.py

```diff
@@ -309,14 +320,17 @@ def validate_submodules(self, config):
 
         for sub_path in config.submodules.exclude:
             path = sub_path.rstrip('/')
+            # TODO: Should we raise an error if the submodule is not found?
             if path in submodules:
                 del submodules[path]
 
         if config.submodules.include != ALL and config.submodules.include:
             submodules_include = {}
             for sub_path in config.submodules.include:
                 path = sub_path.rstrip('/')
-                submodules_include[path] = submodules[path]
+                # TODO: Should we raise an error if the submodule is not found?
+                if path in submodules:
+                    submodules_include[path] = submodules[path]
```

Comment:
Our builds were just failing without a clear error right now :D

Reply:
Yeah, we have a few of these:

* https://github.com/readthedocs/readthedocs.org/issues/10502
* https://github.com/readthedocs/readthedocs.org/issues/9008

We can probably work on them in a different PR.